spanner/ddl/ast.go

```diff
@@ -565,3 +584,60 @@ func maxStringLength(s []string) int {
 	}
 	return n
 }
+
+type Sequence struct {
+	Id               string
+	Name             string
+	SequenceKind     string
+	SkipRangeMin     string
+	SkipRangeMax     string
+	StartWithCounter string
+	ColumnsUsingSeq  map[string][]string
+}
+
+func (seq Sequence) PrintSequence() string {
+	var options []string
+	if seq.SequenceKind != "" {
+		if seq.SequenceKind == "BIT REVERSED POSITIVE" {
+			options = append(options, " sequence_kind='bit_reversed_positive' ")
+		}
+	}
+	if seq.SkipRangeMin != "" {
+		options = append(options, fmt.Sprintf(" skip_range_min = %s ", seq.SkipRangeMin))
+	}
+	if seq.SkipRangeMax != "" {
+		options = append(options, fmt.Sprintf(" skip_range_max = %s ", seq.SkipRangeMax))
+	}
+	if seq.StartWithCounter != "" {
+		options = append(options, fmt.Sprintf(" start_with_counter = %s ", seq.StartWithCounter))
+	}
+
+	seqDDL := fmt.Sprintf("CREATE SEQUENCE %s", seq.Name)
+	if len(options) > 0 {
+		seqDDL += " OPTIONS ( " + strings.Join(options, " , ") + " ) "
+	}
+
+	return seqDDL
+}
+
+func (seq Sequence) PGPrintSequence() string {
+	var options []string
+	if seq.SequenceKind != "" {
+		if seq.SequenceKind == "BIT REVERSED POSITIVE" {
+			options = append(options, " BIT_REVERSED_POSITIVE ")
+		}
+	}
+	if seq.SkipRangeMax != "" && seq.SkipRangeMin != "" {
+		options = append(options, fmt.Sprintf(" SKIP RANGE %s %s ", seq.SkipRangeMin, seq.SkipRangeMax))
+	}
+	if seq.StartWithCounter != "" {
+		options = append(options, fmt.Sprintf(" START COUNTER WITH %s ", seq.StartWithCounter))
+	}
+
+	seqDDL := fmt.Sprintf("CREATE SEQUENCE %s", seq.Name)
+	if len(options) > 0 {
+		seqDDL += strings.Join(options, " ")
```

Comment:
nit: we are adding triple spacing between options like this. Instead, we could remove the the trailing spaces from the Sprintfs  since Join would anyways add a space in between.

Reply:
Sure makes sense